src/Stryker.Core/Stryker.Core/DiffProviders/GitDiffProvider.cs

```diff
@@ -0,0 +1,64 @@
+﻿using LibGit2Sharp;
+using Microsoft.Extensions.Logging;
+using Stryker.Core.Exceptions;
+using Stryker.Core.Logging;
+using Stryker.Core.Options;
+using System.Collections.ObjectModel;
+using System.IO;
+
+namespace Stryker.Core.DiffProviders
+{
+    public class GitDiffProvider : IDiffProvider
+    {
+        private readonly StrykerOptions _options;
+        private readonly ILogger _logger;
+
+        public GitDiffProvider(StrykerOptions options)
+        {
+            _options = options;
+            _logger = ApplicationLogging.LoggerFactory.CreateLogger<GitDiffProvider>();
+        }
+
+        public DiffResult ScanDiff()
+        {
+            if (!_options.DiffEnabled)
+            {
+                return null;
+            }
+            string repositoryPath = Repository.Discover(_options.BasePath)?.Split(".git")[0];
+
+            if (string.IsNullOrEmpty(repositoryPath))
+            {
+                _logger.LogWarning("Could not locate git repo. Unable to determine git diff to filter mutants.");
+            }
+            var diffResult = new DiffResult()
+            {
+                ChangedFiles = new Collection<string>(),
+                TestsChanged = false
+            };
+
+            // A git repository has been detected, calculate the diff to filter
+            using (var repository = new Repository(repositoryPath))
```

Comment:
What happens if there is no git repo?

Reply:
We should probably cancel out of gitdiff if we cannot find a git repo to make sure we don't unexpectedly crash due to user error.

Reply:
As we also do if the given branch cannot be found